docs/proposals/20221611-workload-identity-integration.md

```diff
@@ -0,0 +1,424 @@
+- Supports K8s clusters hosted in any cloud or on premise.
+- Supports both Linux and Windows workloads and removes the need for CRD and pods that intercept IMDS traffic.
+
+From CAPZ perspective
+- The AAD pod identity pod has to be deployed on all the nodes where the capz pod can be potentially scheduled.
+- It has a dependency with the CNI and requires modifying iptables on the node.
+- AAD pod identity is now deprecated. 
+
+To learn more about AZWI please visit this link https://azure.github.io/azure-workload-identity/docs/introduction.html
+
+### <a name='Goals'></a>Goals
+
+- Enable CAPZ pod to use AZWI on the management cluster to authenticate to Azure to create/update/delete resources as part of workload cluster lifecycle management
+
+### <a name='NonGoals'></a>Non Goals
+
+- Automation for migration from AAD pod identity to workload identity.
+- More flexible way of key distribution so that key pair other than the one present for the exiting cluster via `<cluster-name>-sa` secret can be used. To understand more on this please refer [MigrationPlan](#migration-plan)
+- Install workload identity on workload clusters created by CAPZ.
+- Use workload identity for cloud provider azure once supported. 
+
+## <a name='Proposal'></a>Proposal
+
+In AZWI model, Kubernetes cluster itself becomes the token issuer issuing tokens to Kubernetes service accounts. These service accounts can be configured to be trusted on Azure AD application or user assigned managed identity. Workloads/pods can use this service account token which is projected to it's volume and can exchange the projected service account token for an Azure AD token.
+
+The first step for creating a Kubernetes cluster using capz is creating a bootstrap cluster and then deploying capi, capz and other helper components. On a high level the workflow looks the following to be able to use AZWI and create a management cluster.
+
+**Bootstrap Cluster**
+
+- The operator/admin generates signing key pair or BYO key pair. 
+
+- The bootstrap cluster is configured with appropriate flags on kube-apiserver and kube-controller-manager and the key pairs are mounted on the container path for control plane node. See [this](#set-service-account-signing-flags) section for more details on this.
+	- kube-apiserver flags
+		- --service-account-issuer
+		- --service-account-signing-key-file
+		- --service-account-key-file
+	- kube-controller-manager
+		- --service-account-private-key-file
+
+- The operator/admin uploads the following two documents in a blob storage container. These documents is accessible publicy on a URL and this URL is commonly referred as issuer URL in this context. More about this [here](https://azure.github.io/azure-workload-identity/docs/installation/self-managed-clusters/oidc-issuer.html)  
+  - Generate and upload the Discovery Document.
+  - Generate and upload the JWKS Document.
+ 
+- The operator/admin installs the AZWI mutating admission webhook on the bootstrap cluster.
```

Comment:
Workflow seems weird to me. I'm going to register a service account from my laptop as a federated identity provider in Azure?